core/optaplanner-core-impl/src/main/java/org/optaplanner/core/api/solver/SolverJob.java

```diff
@@ -55,10 +56,11 @@
      * Schedules a {@link ProblemChange} to be processed by the underlying {@link Solver} and returns immediately.
      *
      * @param problemChange never null
+     * @return CompletableFuture<Void> that completes after the best solution containing this change has been consumed.
```

Comment:
```suggestion
     * @return completes after the best solution containing this change has been consumed.
```

Don't see the need for duplicating the type information which is already there.